File: solutions/javascript/rna-transcription/1/rna-transcription.js
const translation = nucleotide => {
    switch (nucleotide) {
        case 'G': return 'C';
        case 'C': return 'G';
        case 'T': return 'A';
        case 'A': return 'U';
        default: return nucleotide;
    }
}

export const toRna = sequence => 
  sequence.split('').map(e => translation(e)).join('');

File: solutions/python/rna-transcription/1/rna_transcription.py
def translate(nucleotide):
    match nucleotide:
        case "" : return nucleotide
        case "C" : return "G"
        case "G" : return "C"
        case "A" : return "U"
        case "T" : return "A"
        case _ : return TypeError("Invalid nucleotide")

def to_rna(dna_strand):
    return ''.join([ translate(nucleotide) for nucleotide in dna_strand ])

